### DISCUSSION TITLE
Why are the source code implementations of some hooks not in the code?

### DISCUSSION THREAD

**Discussion by kimmy-wang** [ether/etherpad-lite#5139] [Q&A]
Why are the source code implementations of some hooks not in the code?

**Comment by rhansen** [comment]
I do not understand the question. Can you rephrase?

**Reply by kimmy-wang** [reply]
The source code of some hooks was not found in the code.

**Reply by rhansen** [reply]
Again, I don't understand the question. How can the code not be found in the code?

**Reply by kimmy-wang** [reply]
How to insert the text in the figure below as a hyperlink into the document, and click on the text to jump to a link.

![screenshot-20210804-093443](https://user-images.githubusercontent.com/19725091/128107875-fb06012a-d301-4bfd-89ce-0e7e68c4fd68.png)

**Reply by rhansen** [reply]
Are you trying to write a plugin to support `<a>` elements? If so, there are already some plugins that might do what you want (or require a little modification): [ep_embedded_hyperlinks2](https://www.npmjs.com/package/ep_embedded_hyperlinks2), [ep_full_hyperlinks](https://www.npmjs.com/package/ep_full_hyperlinks). I have not tried them so I don't know how well they work.
